frontend/src/scenes/experimentation/featureFlagLogic.tsx

```diff
@@ -0,0 +1,135 @@
+                },
+                updateMatchGroup: (state, { index, newRolloutPercentage, newProperties }) => {
+                    if (!state) {
+                        return state
+                    }
+                    const groups = [...state?.filters.groups]
+
+                    if (newRolloutPercentage !== undefined) {
+                        groups[index] = { ...groups[index], rollout_percentage: newRolloutPercentage }
+                    }
+
+                    if (newProperties !== undefined) {
+                        groups[index] = { ...groups[index], properties: newProperties }
+                    }
+
+                    return { ...state, filters: { ...state.filters, groups } }
+                },
+                removeMatchGroup: (state, { index }) => {
+                    if (!state) {
+                        return state
+                    }
+                    const groups = [...state.filters.groups]
+                    groups.splice(index, 1)
+                    return { ...state, filters: { ...state.filters, groups } }
+                },
+            },
+        ],
+    },
+    loaders: ({ values }) => ({
+        featureFlag: [
+            null,
+            {
+                loadFeatureFlag: async () => {
+                    if (values.featureFlagId && values.featureFlagId !== 'new') {
+                        return await api.get(`api/feature_flag/${values.featureFlagId}`)
+                    }
+                    return NEW_FLAG
+                },
+                saveFeatureFlag: async (updatedFlag: Partial<FeatureFlagType>) => {
+                    if (!updatedFlag.id) {
+                        return await api.create('api/feature_flag', {
+                            ...updatedFlag,
+                            id: undefined,
+                        })
+                    } else {
+                        return await api.update(`api/feature_flag/${updatedFlag.id}`, {
+                            ...updatedFlag,
+                            id: undefined,
+                        })
+                    }
+                },
+            },
+        ],
+    }),
+    listeners: {
+        saveFeatureFlagSuccess: () => {
+            toast.success(
+                <div>
+                    <h1>Your feature flag has been saved!</h1>
+                    <p>Click here to back to the feature flag list.</p>
+                </div>,
+                {
+                    onClick: () => {
+                        router.actions.push('/feature_flags')
+                    },
+                    closeOnClick: true,
+                }
+            )
+        },
+        deleteFeatureFlag: async ({ featureFlag }) => {
+            deleteWithUndo({
+                endpoint: 'feature_flag',
+                object: { name: featureFlag.name, id: featureFlag.id },
+                callback: () => {
+                    router.actions.push('/feature_flags')
```

Comment:
@macobo I could use your recommendation with a best practice here. Basically the problem is that this same `callback` is called after a user clicks undo. If the user does this, they'll actually be in the feature flag list already, so what we actually would need is to reload the feature flags (i.e. `featureFlagsLogic.actions.loadFeatureFlags()`). The problem is I can't call that action here because when first deleting, the featureFlagsLogic isn't mounted.


I think I would need a way here to either force a reload of the featureFlagsLogic (to retrigger the `loadFeatureFlags()`) or a way to check if the logic is mounted to only call the action in that case. Any ideas?